### PR TITLE
[INJIWEB-977]: add unit testing for Inji Web Utils

### PR DESCRIPTION
**Pull request by kamlesh012**

### CODE REVIEW

File: inji-web/.env [hunk 1]
```diff
@@ -1 +1,2 @@
-CI=false
\ No newline at end of file
+CI=false
+BASE_URL="http://localhost:3010"
```
Comment (vijay151096) [inline inji-web/.env:2]: Why is this needed here?

File: inji-web/package.json [hunk 2]
```diff
@@ -17,7 +15,6 @@
     "autoprefixer": "^9.8.6",
     "crypto-js": "^4.2.0",
     "i18next": "^23.11.2",
-    "inji-web": "file:",
```
Comment (vijay151096) [inline inji-web/package.json:20]: please remove this in both package and package-lock

File: inji-web/src/__tests__/utils/api.test.tsx [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+  let apiModule: ApiModule;
+  let originalEnv: NodeJS.ProcessEnv;
+
+  beforeAll(() => {
+    originalEnv = process.env;
+    process.env = { ...originalEnv, BASE_URL: 'https://example.com' };
+
+    Object.defineProperty(window, 'location', {
+      value: {
+        origin: 'http://localhost:3010'
```
Comment (vijay151096) [inline inji-web/src/__tests__/utils/api.test.tsx:20]: dont use local env

File: inji-web/src/__tests__/utils/i18n.test.tsx [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+    });
+  });
+
+  afterEach(() => {
+    jest.resetModules();
+    windowSpy.mockRestore();
+    clearReduxState();
+  });
+
+  describe('initialization', () => {
```
Comment (vijay151096) [inline inji-web/src/__tests__/utils/i18n.test.tsx:79]: use standard naming for Describe and Test

Describe usually starts with "Test*"
Describe usually starts with "Check*"

File: inji-web/src/__tests__/utils/i18n.test.tsx [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+      const newLanguage = 'fr';
+      await i18nModule.switchLanguage(newLanguage);
+      
+      expect(mockStorage.setItem).toHaveBeenCalledWith('selectedLanguage', newLanguage);
+      expect(mockI18n.changeLanguage).toHaveBeenCalledWith(newLanguage);
+    });
+  });
+
+  describe('getObjectForCurrentLanguage', () => {
+    it('should return object for current language', () => {
```
Comment (vijay151096) [inline inji-web/src/__tests__/utils/i18n.test.tsx:139]: lets stick with test and not it,to be consistant

File: inji-web/src/__tests__/utils/i18n.test.tsx [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+      expect(i18nModule.isRTL('fr')).toBe(false);
+    });
+
+    it('should return correct direction for language', () => {
+      expect(i18nModule.getDirCurrentLanguage('ar')).toBe('rtl');
+      expect(i18nModule.getDirCurrentLanguage('en')).toBe('ltr');
+    });
+  });
+
+  describe('switching language', () => {
```
Comment (vijay151096) [inline inji-web/src/__tests__/utils/i18n.test.tsx:128]: correct the describe and test description for all use case

File: inji-web/src/test-utils/reduxMockUtils.tsx [hunk 7]
```diff
@@ -0,0 +1,19 @@
+// reduxMockUtils.ts
```
Comment (vijay151096) [inline inji-web/src/test-utils/reduxMockUtils.tsx:1]: Why do we need a separate mockUtil for redux, lets consume the existing mockUtil

File: inji-web/src/utils/storage.ts [hunk 8]
```diff
@@ -1,17 +1,24 @@
...
-            data = JSON.parse(data);
+            try {
+                return JSON.parse(data);
+            } catch (error) {
+                return null;
+            }
+
         }
-        return data;
+        return null;
```
Comment (vijay151096) [inline inji-web/src/utils/storage.ts:22]: This has to return data, why it is returning null ?

File: inji-web/src/__tests__/utils/api.test.tsx (hunk 3 above)
Comment (kamlesh012) [reply inji-web/src/__tests__/utils/api.test.tsx:20]: fixed, using  'https://api.collab.mossip.net' instead.

File: inji-web/src/utils/storage.ts (hunk 8 above)
Comment (kamlesh012) [reply inji-web/src/utils/storage.ts:22]: this is for the case when invalide  json data is entered & json parsing fails.then it should return null. 
if you want i can remove the whole test case .

File: inji-web/package.json (hunk 2 above)
Comment (kamlesh012) [reply inji-web/package.json:20]: I had removed it already, it is not showing on my local system. will see what else can be done..

Comment (vijay151096) [reply inji-web/package.json:20]: @kamlesh012 , replace the package.json and package-lock.json from develop and commit it

File: inji-web/src/__tests__/utils/misc.test.tsx [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+        expect(isObjectEmpty(undefined)).toBe(true);
+        expect(isObjectEmpty({ key: 'value' })).toBe(false);
+    });
+
+    test('getTokenRequestBody should return correct request body', () => {
+        const requestBody = getTokenRequestBody('code', 'verifier', 'issuer', 'credential', 'expiry');
+        expect(requestBody).toEqual({
+            'grant_type': 'authorization_code',
+            'code': 'code',
+            'redirect_uri': 'http://localhost/redirect',
```
Comment (vijay151096) [inline inji-web/src/__tests__/utils/misc.test.tsx:44]: local host is used here.

File: inji-web/src/test-utils/reduxMockUtils.tsx (hunk 7 above)
Comment (kamlesh012) [reply inji-web/src/test-utils/reduxMockUtils.tsx]: Added  todo comment.

**Review by vijay151096 — APPROVED** [review]
lgtm
